crates/emitter/src/control_structures.rs

```diff
@@ -586,13 +636,75 @@ where
     F1: Fn(&mut AstEmitter) -> Result<(), EmitError>,
 {
     pub fn emit(&mut self, emitter: &mut AstEmitter) -> Result<(), EmitError> {
-        emitter
-            .control_stack
-            .open_label(self.name, &mut emitter.emit);
+        emitter.control_stack.open_label(
+            self.name,
+            &mut emitter.emit,
+            self.enclosing_emitter_scope_index,
+        );
 
         (self.body)(emitter)?;
 
         emitter.control_stack.close_label(&mut emitter.emit);
         Ok(())
     }
 }
+
+pub enum NonLocalExitKind {
+    Continue,
+    Break,
+}
+
+pub struct NonLocalExitControl<F1>
+where
+    F1: Fn(&mut AstEmitter, BytecodeOffset),
+{
+    _kind: NonLocalExitKind,
```

Comment:
what's the meaning/purpose of this field?

Reply:
maybe this isnt necessary yet.. it is for the later complexity around NonLocalExits

Reply:
okay, can you add comments?